R/legend.R

```diff
@@ -0,0 +1,70 @@
+#' @export
+addLegend = function(
+  map, position = c('topright', 'bottomright', 'bottomleft', 'topleft'),
+  pal, values, na.label = 'NA', bins = 7, colors, opacity = 0.5, labels,
+  title = deparse(substitute(values))
+) {
```

Comment:
Validate inputs:

* If labels is provided, it must be a vector of the same length as colors
* `missing(pal) == missing(values)`, `missing(colors) == missing(labels)`, `missing(pal) != missing(colors)`
* bins will be ignored if colors and labels are provided